From 2464078e80b2dbb4644c591fb4705828e9ff8369 Mon Sep 17 00:00:00 2001
From: George Zacharia <george.zcharia@gmail.com>
Date: Sat, 21 Mar 2020 18:58:41 +0530
Subject: [PATCH] Workaround an issue with WeatherTile

When using custom icon packs, the weather tile image would not be shown properly, work around by forcing to load image from default icon pack in these cases and not break rest of the things
Change-Id: Icfb8a23c0c5bdea38a89e28016af2c33352e7ce0
Change-Id: If97dbb4462b74e0e9e3e57776bc68bd6e68766cb
---
 .../com/android/internal/util/aicp/OmniJawsClient.java | 10 ++++++++--
 .../src/com/android/systemui/qs/tiles/WeatherTile.java |  2 +-
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/core/java/com/android/internal/util/aicp/OmniJawsClient.java b/core/java/com/android/internal/util/aicp/OmniJawsClient.java
index fc23bf7d103..3ed0c662d46 100644
--- a/core/java/com/android/internal/util/aicp/OmniJawsClient.java
+++ b/core/java/com/android/internal/util/aicp/OmniJawsClient.java
@@ -385,12 +385,18 @@ public class OmniJawsClient {
     }
 
     public Drawable getWeatherConditionImage(int conditionCode) {
+        // Wrapper function allowing to load icon from default package defined as ICON_PACKAGE_DEFAULT, for instance in the weather qs tile where a colored icon rendered improperly
+
+        return getWeatherConditionImagefromPack(conditionCode, false /* Don't force default pack*/);
+    }
+
+    public Drawable getWeatherConditionImagefromPack(int conditionCode, boolean defaultPack) {
         if (!isOmniJawsEnabled()) {
             Log.w(TAG, "Requesting condition image while disabled");
             return null;
         }
-        if (!isAvailableApp(mPackageName)) {
-            Log.w(TAG, "Icon pack no longer available - loading default " + mPackageName);
+        if (!isAvailableApp(mPackageName) || defaultPack) {
+            if (DEBUG) Log.w(TAG, "Icon pack no longer available - loading default " + mPackageName);
             loadDefaultIconsPackage();
         }
         if (mRes == null) {
diff --git a/packages/SystemUI/src/com/android/systemui/qs/tiles/WeatherTile.java b/packages/SystemUI/src/com/android/systemui/qs/tiles/WeatherTile.java
index 77871114275..622807db2bb 100644
--- a/packages/SystemUI/src/com/android/systemui/qs/tiles/WeatherTile.java
+++ b/packages/SystemUI/src/com/android/systemui/qs/tiles/WeatherTile.java
@@ -230,7 +230,7 @@ public class WeatherTile extends QSTileImpl<BooleanState> implements OmniJawsCli
                 mWeatherClient.queryWeather();
                 mWeatherData = mWeatherClient.getWeatherInfo();
                 if (mWeatherData != null) {
-                    mWeatherImage = mWeatherClient.getWeatherConditionImage(mWeatherData.conditionCode);
+                    mWeatherImage = mWeatherClient.getWeatherConditionImagefromPack(mWeatherData.conditionCode, true /* force loading from default pack */);
                     mWeatherImage = mWeatherImage.mutate();
                     mWeatherLabel = mWeatherData.temp + mWeatherData.tempUnits;
                 } else {
